backend/utils/passport.ts

```diff
@@ -0,0 +1,53 @@
+import passport from 'passport'
+import passportGitHub from 'passport-github'
+
+import config from './config'
+
+import User, { UserType } from '../types/user'
+
+const GithubStrategy = passportGitHub.Strategy;
+
+passport.serializeUser<UserType, unknown>((user, done) => {
+    done(null, user.id)
+})
+
+passport.deserializeUser((obj, done) => {
+    // test see: https://github.com/passport/express-4.x-facebook-example/blob/master/server.js
+    done(null, obj)
+    
+    // example for mongodb
+    // User.findById(id, (err, user) => {
+    //     done(err, user);
+    // });
+})
+
+
+passport.use(new GithubStrategy({
+    clientID: config.GITHUB_CLIENT_ID ? config.GITHUB_CLIENT_ID : '',
+    clientSecret: config.GITHUB_CLIENT_SECRET ? config.GITHUB_CLIENT_SECRET : '',
```

Comment:
`config.GITHUB_CLIENT_ID ? config.GITHUB_CLIENT_ID : ''` can be written as `config.GITHUB_CLIENT_ID || ''`
More info: [What is short circuit evaluation](https://codeburst.io/javascript-what-is-short-circuit-evaluation-ff22b2f5608c)

Reply:
muutin nämä tässä filussa ja user schemassa